### PR TITLE
ci: move drone to github action

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
**Problem:**
move image related work to github action

**Solution:**
migrate from drone to github

**Related Issue:**
https://github.com/harvester/harvester/issues/5327

**Test plan:**
None

### CODE REVIEW

File: scripts/build [hunk 1]
```diff
@@ -13,8 +13,6 @@ fi
 LINKFLAGS="-X github.com/harvester/node-disk-manager/pkg/version.Version=$VERSION
            -X github.com/harvester/node-disk-manager/pkg/version.GitCommit=$COMMIT $LINKFLAGS"
 
-CGO_ENABLED=0 go build -ldflags "$LINKFLAGS $OTHER_LINKFLAGS" -o bin/node-disk-manager
-if [ "$CROSS" = "true" ] && [ "$ARCH" = "amd64" ]; then
-    GOOS=darwin go build -ldflags "$LINKFLAGS" -o bin/node-disk-manager-darwin
-    GOOS=windows go build -ldflags "$LINKFLAGS" -o bin/node-disk-manager-windows
-fi
```
Comment (brandboat) [inline scripts/build:20]: So we don't need this anymore ? I saw this in node-manager too

Comment (Vicente-Cheng) [reply scripts/build:20]: Yeah, I think so. The Node-disk-manager did not run alone on windows or macos.
Means we do not need these two.

**Review by tserong — APPROVED** [review]
LGTM!

(I wondered why this one didn't add a `.github/workflows/pull_request.yaml` like https://github.com/harvester/harvester-csi-driver/pull/38 has, but then found we've already got `.github/workflows/basic-ci.yaml` which runs on a self-hosed runner, so it looks like the PR case is already handled)

**Comment by Vicente-Cheng** [comment]
> LGTM!
> 
> (I wondered why this one didn't add a `.github/workflows/pull_request.yaml` like [harvester/harvester-csi-driver#38](https://github.com/harvester/harvester-csi-driver/pull/38) has, but then found we've already got `.github/workflows/basic-ci.yaml` which runs on a self-hosed runner, so it looks like the PR case is already handled)

yes, we already have the PR checking CI. It does more than just build, so I did not add `.github/workflows/pull_request.yaml`

**Review by bk201 — APPROVED** [review]
LGTM

**Comment by Vicente-Cheng** [comment]
@mergifyio backport v0.5.x v0.6.x

**Pull request merged**